tools/payments/client.go

```diff
@@ -0,0 +1,185 @@
+package payments
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/brave-intl/bat-go/libs/httpsignature"
+	"github.com/google/uuid"
+	"github.com/redis/go-redis/v9"
+)
+
+const (
+	preparePrefix = "prepare-"
+	submitPrefix  = "authorize-"
+)
+
+var (
+	payout        = strconv.FormatInt(time.Now().Unix(), 10)
+	prepareStream = preparePrefix + payout
+	submitStream  = preparePrefix + payout
+
+	prepareConfigStream = preparePrefix + "-configure-" + payout
+	submitConfigStream  = submitPrefix + "-configure-" + payout
+)
+
+// redisClient is an implementation of settlement client using clustered redis client
+type redisClient struct {
+	env   string
+	redis *redis.ClusterClient
+}
+
+func newRedisClient(ctx context.Context, env string) (*redisClient, error) {
+	tlsConfig := &tls.Config{
+		ServerName: "redis",
+		MinVersion: tls.VersionTLS12,
+		ClientAuth: 0,
+	}
+
+	addrs := strings.Split(os.Getenv("REDIS_ADDRS"), ",")
+	pass := os.Getenv("REDIS_PASS")
+	username := os.Getenv("REDIS_USERNAME")
+
+	// only if environment is local do we hardcode these values
+	if env == "local" {
+		certPool := x509.NewCertPool()
+		pem, err := ioutil.ReadFile("redistest/test/redis/tls/ca.crt")
+		if err != nil {
+			return nil, fmt.Errorf("failed to read test-mode ca.crt: %w", err)
+		}
+		certPool.AppendCertsFromPEM(pem)
+		tlsConfig.RootCAs = certPool
+	}
+
+	rc := &redisClient{
+		env: env,
+		redis: redis.NewClusterClient(
+			&redis.ClusterOptions{
+				Addrs: addrs, Password: pass, Username: username,
+				DialTimeout:     15 * time.Second,
+				WriteTimeout:    5 * time.Second,
+				MaxRetries:      5,
+				MinRetryBackoff: 5 * time.Millisecond,
+				MaxRetryBackoff: 500 * time.Millisecond,
+				PoolSize:        10,
+				PoolTimeout:     30 * time.Second,
+				TLSConfig:       tlsConfig,
+			}),
+	}
+	err := rc.redis.Ping(ctx).Err()
+	if err != nil {
+		return nil, fmt.Errorf("failed to ping redis: %w", err)
+	}
+	return rc, nil
+}
+
+// ConfigureWorker implements settlement client
+func (rc *redisClient) ConfigureWorker(ctx context.Context, stream string, config *WorkerConfig) error {
+	_, err := rc.redis.XAdd(
+		ctx, &redis.XAddArgs{
+			Stream: stream,
+			Values: map[string]interface{}{
+				"data": config}},
```

Comment:
could we use the same message schema for the underlying message i.e put the config into a message

	body, err := json.Marshal(config)
	if err != nil {
		return fmt.Errorf("failed to serialize transaction: %w", err)
	}
	
	config := &prepareWrapper{ // Message
		ID:        uuid.New(),
		Timestamp: time.Now(),
		Body:      string(body),
	}